egs/callhome_diarization/v1/diarization/extract_ivectors.sh

```diff
@@ -0,0 +1,129 @@
+  echo "  --nj <n|10>                                      # Number of jobs (also see num-processes and num-threads)"
+  echo "  --num-threads <n|8>                              # Number of threads for each process"
+  echo "  --stage <stage|0>                                # To control partial reruns"
+  echo "  --num-gselect <n|20>                             # Number of Gaussians to select using"
+  echo "                                                   # diagonal model."
+  echo "  --min-post <min-post|0.025>                      # Pruning threshold for posteriors"
+  exit 1;
+fi
+
+srcdir=$1
+data=$2
+dir=$3
+
+for f in $srcdir/final.ie $srcdir/final.ubm $data/feats.scp ; do
+  [ ! -f $f ] && echo "No such file $f" && exit 1;
+done
+
+
+sub_data=$dir/subsegments_data
+mkdir -p $sub_data
+
+# Set up sliding-window subsegments
+if [ $stage -le 0 ]; then
+  if $hard_min; then
+    awk -v min=$min_segment '{if($4-$3 >= min){print $0}}' $data/segments \
+        > $dir/pruned_segments
+    segments=$dir/pruned_segments
+  else
+    segments=$data/segments
+  fi
+  utils/data/get_uniform_subsegments.py \
+      --max-segment-duration=$window \
+      --overlap-duration=$(echo "$window-$period" | bc) \
+      --max-remaining-duration=$min_segment \
+      --constant-duration=True \
+      $segments > $dir/subsegments
+  utils/data/subsegment_data_dir.sh $data \
+      $dir/subsegments $sub_data
+fi
+
+# Set various variables.
+mkdir -p $dir/log
+sub_sdata=$sub_data/split$nj;
+utils/split_data.sh $sub_data $nj || exit 1;
+
+delta_opts=`cat $srcdir/delta_opts 2>/dev/null`
+
+## Set up features.
+feats="ark,s,cs:add-deltas $delta_opts scp:$sub_sdata/JOB/feats.scp ark:- |"
+
+
+if [ $stage -le 1 ]; then
+  echo "$0: extracting iVectors"
+  dubm="fgmm-global-to-gmm $srcdir/final.ubm -|"
+
+  $cmd JOB=1:$nj $dir/log/extract_ivectors.JOB.log \
+    gmm-gselect --n=$num_gselect "$dubm" "$feats" ark:- \| \
+    fgmm-global-gselect-to-post --min-post=$min_post $srcdir/final.ubm "$feats" \
+       ark,s,cs:- ark:- \| scale-post ark:- $posterior_scale ark:- \| \
+    ivector-extract --verbose=2 $srcdir/final.ie "$feats" ark,s,cs:- \
+      ark,scp,t:$dir/ivector.JOB.ark,$dir/ivector.JOB.scp || exit 1;
+fi
+
+if [ $stage -le 2 ]; then
+  echo "$0: combining iVectors across jobs"
+  for j in $(seq $nj); do cat $dir/ivector.$j.scp; done >$dir/ivector.scp || exit 1;
+  cp $sub_data/{segments,spk2utt,utt2spk} $dir
+fi
+
+if [ $stage -le 3 ]; then
+  echo "$0: Computing mean of iVectors"
+  $cmd $dir/log/mean.log \
+    ivector-mean scp:$dir/ivector.scp $dir/mean.vec || exit 1;
+fi
+
+if [ $stage -le 4 ]; then
+  if [ -z "$pca_dim" ]; then
+    pca_dim=-1
+  fi
+  echo "$0: Computing whitening transform"
+  $cmd $dir/log/transform.log \
+    est-pca --read-vectors=true --normalize-mean=false \
+      --normalize-variance=true --dim=$pca_dim \
+      scp:$dir/ivector.scp $dir/transform.mat || exit 1;
+fi
```

Comment:
TODO: I think we should remove the mean and whitening transform stuff from here, and just compute it in the run.sh, like we do with PLDA.